test/fslock.cpp

```diff
@@ -0,0 +1,152 @@
+#include "fslock.h"
+
+#include <fcntl.h>
+#include <semaphore.h>
+#include <sys/stat.h>
+#include <sys/types.h>
+#include <unistd.h>
+
+#include "3rd-party/catch.hpp"
+#include "test-helpers.h"
+
+using namespace newsboat;
+
+// Forks and calls FsLock::try_lock() in the child process.
+class LockProcess {
+public:
+	LockProcess(std::string lock_location)
+	{
+		sem_start = sem_open(sem_start_name, O_CREAT, 0644, 0);
+		sem_stop = sem_open(sem_stop_name, O_CREAT, 0644, 0);
+
+		pid = ::fork();
+		if (pid == -1) {
+			FAIL("LockProcess: fork() failed");
+		} else if (pid > 0) {
+			// Parent process: Wait until child process has finished calling try_lock
+			sem_wait(sem_start);
+		} else {
+			// Child process: Call try_lock, signal parent to continue, wait for parent process signal to stop
+			pid_t ignore_pid;
+			FsLock lock;
+			lock.try_lock(lock_location, ignore_pid);
+			sem_post(sem_start);
+
+			sem_wait(sem_stop);
+			// Exit directly without running destructors (making sure we don't interfere with the original process)
+			_exit(0);
+		}
+	}
+
+	pid_t get_child_pid()
+	{
+		return pid;
+	}
+
+	~LockProcess()
+	{
+		if (pid > 0) {
+			// Parent process: Signal child process to exit and wait for it to finish
+			sem_post(sem_stop);
+			::waitpid(pid, nullptr, 0);
+
+			REQUIRE(sem_unlink(sem_start_name) == 0);
+			REQUIRE(sem_unlink(sem_stop_name) == 0);
+			REQUIRE(sem_close(sem_start) == 0);
+			REQUIRE(sem_close(sem_stop) == 0);
+		}
+	}
+
+private:
+	const char* sem_start_name = "/newsboat-test-fslock-start";
+	const char* sem_stop_name = "/newsboat-test-fslock-stop";
+
+	pid_t pid;
+	sem_t* sem_start;
+	sem_t* sem_stop;
+};
+
+
+TEST_CASE("try_lock() returns an error if lock-file permissions or location are invalid",
+	"[FsLock]")
+{
+
+	GIVEN("An invalid lock location") {
+		const TestHelpers::TempDir test_directory;
```

Comment:
Wow, I'm kinda ashamed that, despite being so bent on const correctness, I missed this opportunity in all the tests I wrote and reviewed :)